src/Validator/Product.php

```diff
@@ -41,9 +59,47 @@ public function validateRequest(RequestInterface $request): bool
         $urlKey = $this->getPathFrontName($request);
         $productCollection = $this->productCollection->clear();
         $productCollection->addAttributeToFilter('url_key', $urlKey);
+
         $ids = $productCollection->getAllIds();
         $productId = reset($ids);
 
-        return !!$productId;
+        if (!$productId) return false;
+
+        $typeIds = $productCollection->getProductTypeIds();
+        $type = reset($typeIds);
+        $parameters = $request->getParams();
+
+        if ($type === Configurable::TYPE_CODE && !empty($parameters) && !$this->checkConfigurableProduct($productId, $parameters))
+            return false;
+
+        return true;
+    }
+
+    /**
+     * Checks whether configurable product with specified parameters does exist
+     * @param int $productId
+     * @param array $parameters
+     * @return bool
+     */
+    protected function checkConfigurableProduct(int $productId, array $parameters): bool {
+        $product = $this->product->load($productId);
+
+        $attributes = $this->configurable->getConfigurableAttributes($product);
+
+        foreach ($attributes as $attribute) {
+            $attributeCode = $attribute->getProductAttribute()->getAttributeCode();
+
+            if (array_key_exists($attributeCode, $parameters)) {
+                $parameterValue = $parameters[$attributeCode];
+                unset($parameters[$attributeCode]);
+                $options = $attribute->getOptions();
+
+                if (array_search($parameterValue, array_column($options,  'value_index')) === false) return false;
+            }
+        }
+
+        if (!empty($parameters)) return false;
```

Comment:
this check is already in a call stack, and a method can be called in current class or subclass, looks like this line could be easily removed.

Reply:
Please, could you elaborate a bit more on it. The first check in `validateRequest` function prevents executing unnecessary logic execution when there are no parameters passed, as this one, paired with unset, ensures that no other query parameters despite the ones that are used by configurable product were passed.